Sources/Logging.php

```diff
@@ -543,7 +543,7 @@ function logActions(array $logs)
 		if (isset($log['extra']['member_affected']))
 			$memID = $log['extra']['member_affected'];
 		else
-			$memID = $user_info['id'];
+			$memID = (isset($user_info['id']) && !empty($user_info['id']) ? $user_info['id'] : (!empty($log['extra']['member']) ? $log['extra']['member'] : 0));
```

Comment:
!empty implies isset, you don't need to specify both.  As another note.  If the value being checked is set to null, it is considered empty and not set, you have to use is_null checks in that case, except this doesn't work for array keys, you have to use array_key_exists.  We technically could use null coalescing here as well.  That is a PHP 7.0 feature.

Reply:
I think is because I initially had the condition reversed.
$log['extra']['member'] might not need to be checked since it's already checked.

'if (isset($log['extra']['member']) && !is_numeric($log['extra']['member']))'

Perhaps like this?
`$memID = $user_info['id'] ?? $log['extra']['member']`

Reply:
2.1 does say 7.0 or higher, so I would say yes, just like that for null coalescing. Could even go further:

```$memID = $user_info['id'] ?? $log['extra']['member'] ?? 0;```

Reply:
I updated it.